### DISCUSSION TITLE
504: GATEWAY_TIMEOUT

### DISCUSSION THREAD

**Discussion by VijenSingh** [vercel/next.js#68250] [Help]
### Summary

Hi 
I am getting error 504: GATEWAY_TIMEOUT
Code: FUNCTION_INVOCATION_TIMEOUT
ID: bom1::cdntf-1722188554581-8ae2c34e2fa4

### Additional information

```js
here is my link https://google-tracking-ochre.vercel.app/
when I click go to track it goes to https://google-tracking-ochre.vercel.app/track

and give me an error 

please help me
```

**Comment by icyJoseph** [comment]
🤔 deploying to Vercel is "serverless", so every time a request hits the server, the entire express server is set up, and listeners fixed, and then only one of them is executed. 

Because it is "serverless", there's a time budget, in hobby plans it is about 10 seconds IIRC. 

So if your track handler, requires some seconds to complete, added to the boot-up time... you are likely to go over your time budget.

In the Vercel dashboard, you can find logs for your functions, try to see there what's going on. 

Try to benchmark, locally, how long it takes for a request to do a roundtrip.
